test/mechanics/cell/PatchTest.cpp

```diff
@@ -282,29 +284,11 @@ BOOST_AUTO_TEST_CASE(PatchTestDispl)
     auto gradient = assembler.BuildVector(cellGroup, {displ}, GradientF);
     auto hessian = assembler.BuildMatrix(cellGroup, {displ}, Hessian0F);
 
-    Eigen::MatrixXd kJJ = Eigen::MatrixXd(hessian.JJ(displ, displ));
-    Eigen::MatrixXd kJK = Eigen::MatrixXd(hessian.JK(displ, displ));
-    Eigen::MatrixXd kKJ = Eigen::MatrixXd(hessian.KJ(displ, displ));
-    Eigen::MatrixXd kKK = Eigen::MatrixXd(hessian.KK(displ, displ));
-    BOOST_TEST_MESSAGE("hessian JJ \n" << kJJ);
-    BOOST_TEST_MESSAGE("hessian JK \n" << kJK);
-    BOOST_TEST_MESSAGE("hessian KJ \n" << kKJ);
-    BOOST_TEST_MESSAGE("hessian KK \n" << kKK);
-
-
-    BOOST_TEST_MESSAGE("GradientJ \n" << gradient.J);
-    BOOST_TEST_MESSAGE("GradientK \n" << gradient.K);
-
-    //      have a look at DISS_UNGER, page 28 for all that CMat stuff.
-    Eigen::MatrixXd Kmod = kJJ - CMat.transpose() * kKJ - kJK * CMat + CMat.transpose() * kKK * CMat;
-    Eigen::VectorXd Rmod = gradient.J[displ] - CMat.transpose() * gradient.K[displ];
-    Eigen::VectorXd RmodConstrained = (kJK - CMat.transpose() * kKK) * (-constraints.GetRhs(displ, 0));
-
-    Eigen::VectorXd newDisplacementsJ = Kmod.ldlt().solve(Rmod + RmodConstrained);
-    Eigen::VectorXd newDisplacementsK = -CMat * newDisplacementsJ + constraints.GetRhs(displ, 0);
+    int numIndependentDofs = dofInfo.numIndependentDofs[displ];
```

Comment:
If we add this numIndependentDofs to the dof-class (or better the total number of dofs for this type), then this could be calculated directly.